server/cluster/coordinator.go

```diff
@@ -103,6 +103,27 @@ func (c *coordinator) patrolRegions() {
 			return
 		}
 
+		// Check suspect regions first.
+		for _, id := range c.cluster.GetSuspectRegions() {
```

Comment:
it seems to be similar to the block below, can we abstract it into a function to avoid redundancy

Reply:
em, I think they are not quite alike. In this loop, the region may not exist yet, and we need to remove the id from suspect list in some cases.